tests/test_search_readonly.py

```
@@ -153,3 +189,557 @@ def test_readonly_mode_blocked_even_when_menu_item_bypassed(self, tmp_path):
+    junctions) outright on a read-only install, not follow them — a link
+    under a permitted folder could target the excluded ITAR directory
+    (CodeRabbit, PR #315).
+
+    Mocks _is_reparse_point() directly rather than creating a real
+    symlink/junction, since that requires Developer Mode or elevation on
+    Windows and would otherwise skip on many CI runners — see
+    TestReparsePointExclusionOnReadonly below for a real-symlink test that
+    skips gracefully where unsupported.
+    """
+
+    def _context(self, tmp_path: Path, *, readonly_mode: bool, settings: dict = None) -> AppContext:
```

Ruff (0.16.1)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)